packages/sonicwall/data_stream/firewall/elasticsearch/ingest_pipeline/default.yml

```
@@ -1,72 +1,503 @@
----
-description: Pipeline for Sonicwall-FW
-
 processors:
-  # ECS event.ingested
+  - trim:
+      field: message
+  - rename:
+      field: message
+      target_field: event.original
+  - set:
+      field: event.module
+      value: sonicwall
   - set:
       field: event.ingested
       value: '{{_ingest.timestamp}}'
+  - set:
+      field: event.dataset
+      value: sonicwall.firewall
   - set:
       field: ecs.version
-      value: '8.0.0'
-  # User agent
-  - user_agent:
-      field: user_agent.original
+      value: 8.0.0
+  - grok:
+      field: event.original
+      patterns:
+        - '^%{IP:event.provider:IP} +%{GREEDYDATA:message}$'
+        - >-
+          ^%{DATA}%{TIMESTAMP:event.provided} %{IP:event.provider:IP}
+          %{GREEDYDATA:message}$
```

I don't think the multi-line string improves clarity here and is arguably more fragile to un-noticed damage in the future.

```suggestion
      patterns:
        - "^%{DATA}%{TIMESTAMP:event.created} %{IP:observer.ip:ip} %{GREEDYDATA:message}$"
        - "^%{DATA}%{IP:observer.ip:ip} %{GREEDYDATA:message}$"
```

---

Missed that, I just used a json > yaml converter from doing testing in kibana. Looks like that added that.

Just had to add an extra '+' to keep the message field from starting with a whitespace and breaking the kv proccessor